transport/twirp/server.go

```diff
@@ -78,10 +81,20 @@ func (s Server) ServeTwirp(ctx context.Context, req interface{}) (context.Contex
 			s.finalizer(ctx, req)
 		}()
 	}
-
+	// Extract the headers from the ctx
+	var (
+		reqHeader http.Header
+		ok        bool
+	)
+	reqHeader, ok = twirp.HTTPRequestHeaders(ctx)
```

Comment:
I think it is because you have to move headers from the request to the context as explained in https://twitchtv.github.io/twirp/docs/headers.html#read-http-headers-from-requests